### PR TITLE
Shorten stackex link timedelta

### PR DESCRIPTION
**Pull request by joshr17**
Main change is stackex timestamp `356 * 2 -> 365 // 4 `. ID-GNN performs well on these link tasks.

### CODE REVIEW

File: examples/gnn_link.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 parser.add_argument("--dataset", type=str, default="rel-hm")
 parser.add_argument("--task", type=str, default="rel-hm-rec")
 parser.add_argument("--lr", type=float, default=0.001)
-parser.add_argument("--epochs", type=int, default=20)
+parser.add_argument("--epochs", type=int, default=5)
```
Comment (weihua916) [inline examples/gnn_link.py:30]: Why do we shorten it? Is it enough to converge?

File: relbench/external/loader.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ class SparseTensor:
     def __init__(
         self,
         sparse_tensor: Tensor,
-        device: torch.device | None = None,
+        device: Union[torch.device, None] = None,
```
Comment (weihua916) [inline relbench/external/loader.py:47]: Why? The current one is a recommended type annotation for Python 3.10.

File: relbench/tasks/stackex.py [hunk 3]
```diff
@@ -361,7 +361,7 @@ class UsersInteractTask(RelBenchLinkTask):
     timedelta = pd.Timedelta(days=365 // 4)
 
     metrics = [link_prediction_precision, link_prediction_recall, link_prediction_map]
-    eval_k = 10
+    eval_k = 100
```
Comment (weihua916) [inline relbench/tasks/stackex.py:364]: How does this affect our MAP metric?

**Review by weihua916 — APPROVED** [review]

File: relbench/tasks/stackex.py (hunk 3 above)
Comment (joshr17) [reply relbench/tasks/stackex.py:364]: The difference in mAP is:

with `eval_k=100`

- ID-GNN `0.03751`
- global-popularity `0.12440`

with `eval_k=10`

- ID-GNN `0.01780`
- global-popularity `0.11706`

Change was made simply to bring this task into line with the other two stackex link tasks, which also use `eval_k=100`.

File: relbench/external/loader.py (hunk 2 above)
Comment (joshr17) [reply relbench/external/loader.py:47]: I can change back - i am usign Python 3.9 so this was throwing an error. Should we be aware of this kind of back-compatability?

File: examples/gnn_link.py (hunk 1 above)
Comment (joshr17) [reply examples/gnn_link.py]: Reverted

File: relbench/tasks/stackex.py (hunk 3 above)
Comment (weihua916) [reply relbench/tasks/stackex.py]: Makes sense! thanks

File: relbench/external/loader.py (hunk 2 above)
Comment (weihua916) [reply relbench/external/loader.py]: Got it, if it throws an error in your env, feel free to change it. I think many people are still using 3.9

**Pull request merged**